File: examples/lh-server/python/07-child-workflows/workflow.py
from littlehorse.workflow import Workflow, WorkflowThread


def child_logic(wf: WorkflowThread) -> None:
    name = wf.declare_str("name").required()
    wf.complete(wf.execute("greet", name))


def parent_logic(wf: WorkflowThread) -> None:
    input_name = wf.declare_str("input-name").required()
    child_output = wf.declare_str("child-output")
    child = wf.run_wf("greeting-child", inputs={"name": input_name})
    wf.execute("greet", "hi from parent")
    child_output.assign(wf.wait_for_child_wf(child))


def get_workflows() -> tuple[Workflow, Workflow]:
    return (
        Workflow("greeting-child", child_logic),
        Workflow("greeting-parent", parent_logic),
    )

File: examples/lh-server/go/07-child-workflows/workflow.go
package main

import "github.com/littlehorse-enterprises/littlehorse/sdk-go/littlehorse"

func ChildWorkflow(wf *littlehorse.WorkflowThread) {
	name := wf.DeclareStr("name").Required()
	wf.Complete(wf.Execute("greet", name))
}

func ParentWorkflow(wf *littlehorse.WorkflowThread) {
	inputName := wf.DeclareStr("input-name").Required()
	childOutput := wf.DeclareStr("child-output")
	child := wf.RunWf("greeting-child", map[string]interface{}{"name": inputName})

	wf.Execute("greet", "hi from parent")
	childOutput.Assign(wf.WaitForChildWf(child))
}
